### PR TITLE
Sligthly more coherent raw message interfaces.

### PR DESCRIPTION
**Pull request by gbin**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -216,14 +208,18 @@ pub struct MAVLinkAnyMessageRaw<const S: usize>([u8; S]);
 
 /// This extracts all the common useful fields from a MAVLink message be a V1 or V2
 #[allow(clippy::len_without_is_empty)]
-pub trait CommonMessageRaw {
+pub trait CommonMessageRaw
+where
+    Self: Sized,
```
Comment (massi31415) [inline src/lib.rs:213]: nit: Maybe `Sized + Copy` too no?

File: src/lib.rs [hunk 2]
```diff
@@ -256,14 +252,19 @@ impl CommonMessageRaw for MAVLinkV1MessageRaw {
 
     #[inline]
     fn len(&self) -> usize {
-        1 + HEADER_SIZE_V1 + self.payload_length()
+        1 + HEADER_SIZE_V1 + self.payload_length() + 2
```
Comment (massi31415) [inline src/lib.rs:255]: ahh there was a bug before?

Comment (massi31415) [reply src/lib.rs:255]: those are the two trailing zeros right?

File: src/lib.rs [hunk 3]
```diff
@@ -346,6 +347,10 @@ impl MAVLinkV1MessageRaw {
         ]);
 
         let crc = self.calculate_crc::<M>();
+        println!("CRC = {:x}", crc);
+        println!("Payload len = {:x} ({})", payload_len, payload_len);
+        println!("Payload offset = {}", 1 + HEADER_SIZE_V1 + payload_len);
```
Comment (massi31415) [inline src/lib.rs:352]: debug prints?

Comment (massi31415) [reply src/lib.rs:352]: no need for that I think

**Review by massi31415 — APPROVED** [review]
left nit comments but lgtm

File: src/lib.rs (hunk 2 above)
Comment (gbin) [reply src/lib.rs:255]: this is the CRC :facepalm:

File: src/lib.rs (hunk 3 above)
Comment (gbin) [reply src/lib.rs:352]: will get removed thanks

**Pull request closed without merging**